apps/src/sites/code.org/pages/views/workshop_search.js

```diff
@@ -10,30 +11,265 @@ const markerCustererOptions = {
+    loadGoogleMapWorkshops();
+  }
 });
 
-function initializeMap() {
+function initializeMapboxMap() {
+  map = new mapboxgl.Map({
+    container: 'mapbox-map',
+    style: 'mapbox://styles/mapbox/streets-v11',
+    center: [-95.665, 37.6],
+    zoom: 3
+  });
+
+  map.dragRotate.disable();
+  map.scrollZoom.disable();
+  map.dragPan.disable();
+  map.addControl(
+    new mapboxgl.NavigationControl({showCompass: false}),
+    'bottom-right'
+  );
+  const enableMouseControls = () => {
+    map.scrollZoom.enable();
+    map.dragPan.enable();
+  };
+  // Enable mouse controls when the map is clicked
+  map.on('click', enableMouseControls);
+  // Enable mouse controls when the zoom (+/-) buttons are pressed
+  map.on('zoom', enableMouseControls);
+
+  var geocoder = new MapboxGeocoder({
+    accessToken: mapboxgl.accessToken,
+    mapboxgl: mapboxgl,
+    marker: false,
+    types: 'country,region,district,postcode,locality,place'
+  });
+  document.getElementById('mapboxgeocoder').appendChild(geocoder.onAdd(map));
+}
+
+function loadMapboxWorkshops() {
+  const deepDiveOnly = $('#properties').attr('data-deep-dive-only');
+  let url = '/dashboardapi/v1/pd/k5workshops?geojson=1';
+  if (deepDiveOnly !== undefined) {
+    url += '&deep_dive_only=1';
+  }
+  map.addSource('workshops', {
+    type: 'geojson',
+    data: url,
+    cluster: true,
+    clusterRadius: 20,
+    clusterMaxZoom: 12,
+    clusterProperties: {
+      clustered_workshop_count: ['+', ['get', 'workshop_count']]
+    }
+  });
+
+  placeClusters();
+  placeIntroWorkshops();
+  placeDeepDiveWorkshops();
+}
+
+function placeClusters() {
+  map.addLayer({
+    id: 'workshop-clusters',
+    type: 'circle',
+    source: 'workshops',
+    filter: ['has', 'point_count'],
+    paint: {
+      'circle-color': colors['teal'],
+      'circle-radius': 20,
+      'circle-blur': 0.75
+    }
+  });
+  map.addLayer({
+    id: 'workshop-clusters-count',
+    type: 'symbol',
+    source: 'workshops',
+    filter: ['has', 'point_count'],
+    layout: {
+      'text-field': ['get', 'clustered_workshop_count'],
+      'text-font': ['DIN Offc Pro Medium', 'Arial Unicode MS Bold'],
+      'text-size': 12,
+      'text-allow-overlap': true
+    }
+  });
+  map.on('click', 'workshop-clusters', function(e) {
+    var features = map.queryRenderedFeatures(e.point, {
+      layers: ['workshop-clusters']
+    });
+    var clusterId = features[0].properties.cluster_id;
+    map
+      .getSource('workshops')
+      .getClusterExpansionZoom(clusterId, function(err, zoom) {
+        if (err) {
+          console.log(err);
+          return;
+        }
+
+        map.easeTo({
+          center: features[0].geometry.coordinates,
+          zoom: zoom
+        });
+      });
+  });
+  map.on('mouseenter', 'workshop-clusters', function() {
+    map.getCanvas().style.cursor = 'pointer';
+  });
+}
+
+function placeIntroWorkshops() {
+  map.loadImage(
+    'https://maps.google.com/mapfiles/kml/paddle/red-blank.png',
```

Comment:
To state the obvious, we should update this :)

Reply:
Agreed! I don't think this blocks this PR but let me know if you disagree!